bootloader/src/pyi_utils.c

```diff
@@ -1028,159 +1057,279 @@ pyi_utils_create_child(const char *thisfile, const ARCHIVE_STATUS* status,
+            ProcessSerialNumber psn;
+            AppleEvent evtCopy;
+            AEAddressDesc target;
+            OSStatus err = noErr;
+            AEEventClass evtClass;
+            AEEventID evtID;
 
+            if (apple_event_is_open_uri) {
+                evtClass = kInternetEventClass;
+                evtID = kAEGetURL;
+            } else {
+                /* open doc */
+                evtClass = kCoreEventClass;
+                evtID = kAEOpenDocuments;
+            }
 
-  return (err);
+            VS("LOADER [AppleEvent EVT_FWD]: Will forward %s event to child pid %ld...\n", descStr, (long)child_pid);
+
+            err = GetProcessForPID(child_pid, &psn);
+            if (err != noErr) return (OSErr)err;
+            VS("LOADER [AppleEvent EVT_FWD]: Creating target desc.\n");
+            /* re-target the event to our child process: creating a descriptor and event object for it */
+            err = AECreateDesc(typeProcessSerialNumber, &psn, sizeof(psn), &target);
+            if (err != noErr) return (OSErr)err;
+            VS("LOADER [AppleEvent EVT_FWD]: Creating dupe event.\n");
+            err = AECreateAppleEvent(evtClass, evtID, &target, kAutoGenerateReturnID, kAnyTransactionID, &evtCopy);
+            if (err != noErr) goto cleanup2;
+
+            { /* <-- These curly braces aren't a typo. We are creating a nested scope here for below vars. */
+                char buf[64*1024]; /* 64 KiB buffer should be enough for drag/drops */
+                DescType actualType = 0;
+                Size actualSize = 0;
+                VS("LOADER [AppleEvent EVT_FWD]: Getting param.\n");
+                err = AEGetParamPtr(theAppleEvent, keyDirectObject, typeWildCard, &actualType, buf, sizeof(buf),
+                                    &actualSize);
+                if (err != noErr) goto cleanup1;
+                VS("LOADER [AppleEvent EVT_FWD]: Got param type=%x (%s) size=%ld\n",
+                   actualType, CC2Str(actualType), (long)actualSize);
+                VS("LOADER [AppleEvent EVT_FWD]: Putting param.\n");
+                err = AEPutParamPtr(&evtCopy, keyDirectObject, actualType, buf, actualSize);
+                if (err != noErr) goto cleanup1;
+            }
+            VS("LOADER [AppleEvent EVT_FWD]: Sending message...\n");
+            err = AESendMessage(&evtCopy, NULL, kAENoReply, 60 /* 60 = about 1.0 seconds timeout */);
+            VS("LOADER [AppleEvent EVT_FWD]: Handler forwarded %s message to child pid %ld.\n",
+               descStr, (long)child_pid);
+
+        cleanup1:
+            AEDisposeDesc(&evtCopy); /* dispose apple event copy */
+        cleanup2:
+            AEDisposeDesc(&target);
+            if (err)
+                VS("LOADER [AppleEvent EVT_FWD]: OpenDocument handler got error %d\n", (int)err);
+            return (OSErr)err;
+        }
+    } else {
+        /* Not GetURL or OpenDoc, so just ignore */
+        VS("LOADER [AppleEvent]: Handler ignoring event.\n");
```

Comment:
Do you think it would be valuable to log the event to ease debugging or future new event handling?

Reply:
Yes, good suggestion. Will add.

Reply:
Added.